### PR TITLE
Implemented drag and select property to tt

### PR DESCRIPTION
**Pull request by lameinthebox**
You can Ctrl-Drag or Ctrl-Click timeslots to toggle selected timeslots.

- Before
screenshot from 2018-06-06 14-10-38

- After
screenshot from 2018-06-06 14-10-24

### CODE REVIEW

**Comment by lameinthebox** [comment]
*Update*

I've re-implemented rendering method from table to canvas. None of the interfaces is changed.
You can click on a lecture to see the corresponding popup.
You can Ctrl-click or Ctrl-drag to select timeslots.

screenshot from 2018-06-07 09-47-20

screenshot from 2018-06-07 09-47-42

**Comment by Tommybro** [comment]
강의들을 불투명하게 해서 밑에 선들이 안보이게 해주세요

**Comment by lameinthebox** [comment]
보이는 게 이뻐서 일부러 이렇게 했는뎅 ㅜㅜ

**Comment by Tommybro** [comment]
안보이는게 이쁜뎅 ㅠㅠ

**Comment by lameinthebox** [comment]
Modified code as you've requested.

screenshot from 2018-06-07 10-08-57

**Comment by Tommybro** [comment]
GOOD

File: src/components/molecules/LecturePopup/index.js [hunk 1]
```diff
@@ -57,6 +57,10 @@ class LecturePopup extends React.Component {
     this.props.onToggleLikeIt(this.props.lecture.id, isAdd, evaluation.id)
   }
 
+  callBack = (datafromchild) => {
+      console.log(datafromchild)
```
Comment (Tommybro) [inline src/components/molecules/LecturePopup/index.js]: Please modify `datafromchild` to `dataFromChild`
And I think this function does not need.

Comment (lameinthebox) [reply src/components/molecules/LecturePopup/index.js]: 무슨 소리지?
인자가 필요 없다고?

File: src/components/molecules/TTRenderer/index.js [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+const blockWidth = 130
+const blockHeight = 29
+const canvasWidth = (blockWidth + 2) * 7
+const canvasHeight = (blockHeight + 2) * 25
+
+class TTRenderer extends React.Component {
+  constructor(props) {
+    super(props)
+    // console.log('props:', props)
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: Please send pull request after remove useless code from now on

File: src/components/molecules/TTRenderer/index.js [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        [0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0],
+      ],
+    }
+    // console.log(this.state)
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: remove this code

File: src/components/molecules/TTRenderer/index.js [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+class TTRenderer extends React.Component {
+  constructor(props) {
+    super(props)
+    // console.log('props:', props)
+    this.state = {
+      lectures: props.lectures,
+      startPoint: null,
+      endPoint: null,
+      blocks: [
+        [0, 0, 0, 0, 0, 0],
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: Um.. this can be done like this.
You write this code out of class
```js
const blocks = []
for (int i = 0; i < 24; i += 1) {
  blocks.push([0, 0, 0, 0, 0, 0])
}
```
And then at `this.state`
```js
blocks: [...blocks],
```

File: src/components/molecules/LecturePopup/index.js (hunk 1 above)
Comment (holenet) [reply src/components/molecules/LecturePopup/index.js]: @lameinthebox 
I think @Tommybro means that 'Logging is not necessary, so please delete it'.

File: src/components/molecules/TTRenderer/index.js [hunk 5]
```diff
@@ -0,0 +1,416 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+
+import LecturePopup from '../../../containers/LecturePopup'
+
+
+const blockWidth = 130
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: Could you change the code so that the timetable fill the parent container(component)?
That means, there should be no space no matter what value the screen width have.

File: src/components/molecules/TimeTable/index.js [hunk 6]
```diff
@@ -104,7 +19,7 @@ class TimeTable extends React.Component {
     searchOpen: false,
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentWillReceiveProps (nextProps) {
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please remove the space between function name and parameter parentheses.

File: src/components/molecules/TimeTable/index.js [hunk 7]
```diff
@@ -118,7 +33,7 @@ class TimeTable extends React.Component {
       console.log('There is no TimeTable')
       return
     }
-    this.props.onModifyContent({ memo: this.state.memoInput })
+    this.props.onModifyContent({memo: this.state.memoInput})
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please add spaces after opening brace and before closing brace.

File: src/components/molecules/TimeTable/index.js [hunk 8]
```diff
@@ -129,7 +44,7 @@ class TimeTable extends React.Component {
       console.log('There is no TimeTable')
       return
     }
-    this.props.onModifyContent({ title: this.state.titleInput })
+    this.props.onModifyContent({title: this.state.titleInput})
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Same to here.

File: src/components/molecules/TimeTable/index.js [hunk 9]
```diff
@@ -152,74 +67,33 @@ class TimeTable extends React.Component {
     this.props.onSend(sendInfo)
   }
 
-  handleChange = (e, { name, value }) => {
-    this.setState({ [name]: value })
+  handleChange = (e, {name, value}) => {
+    this.setState({[name]: value})
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Same here x2.

File: src/components/molecules/TimeTable/index.js [hunk 10]
```diff
@@ -152,74 +67,33 @@ class TimeTable extends React.Component {
...
-      return null
-    }
-
-    return (<td key={key} style={{ border: '1px solid black', width: '100px', height: '30px' }} />)
-  }
-
-  createRow = (time) => {
-    return ['월', '화', '수', '목', '금', '토'].map((day) => this.createTimeSlot(day, time))
+  getBlocks = (blocks) => {
+    // console.log(this)
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please remove this line.

File: src/components/molecules/TimeTable/index.js [hunk 11]
```diff
@@ -152,74 +67,33 @@ class TimeTable extends React.Component {
...
-      }
-      return null
-    }
-
-    return (<td key={key} style={{ border: '1px solid black', width: '100px', height: '30px' }} />)
-  }
-
-  createRow = (time) => {
-    return ['월', '화', '수', '목', '금', '토'].map((day) => this.createTimeSlot(day, time))
+  getBlocks = (blocks) => {
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: This function seems unnecessary...

File: src/components/molecules/TimeTable/index.js [hunk 12]
```diff
@@ -152,74 +67,33 @@ class TimeTable extends React.Component {
...
-          ))}
-        </tbody>
-      </table>
+      <div>
+        <TTRenderer
+          onChange={this.getBlocks}
+          lectures={this.props.lectures}
+          deleteLecture={this.props.onDeleteLecture}
+          addToNotRecommends={this.props.onAddToNotRecommends}
+          getEvaluations={this.props.onGetEvaluations}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please add the prefix 'on' to the names of callback functions(above 3).

File: src/components/molecules/TimeTable/index.js [hunk 13]
```diff
@@ -152,74 +67,33 @@ class TimeTable extends React.Component {
...
+          notRecommends={this.props.notRecommends}
+          canModify={this.props.canModify}
+        />
+      </div>
     )
   }
 
-  render() {
+  render () {
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please remove this space.

File: src/components/molecules/TimeTable/index.js [hunk 14]
```diff
@@ -231,16 +105,16 @@ class TimeTable extends React.Component {
...
                   {this.props.canModify &&
                   <Popup
                     trigger={<button
                       className="ui icon button"
-                      onClick={() => this.setState({ isModifyingTitle: true })}
+                      onClick={() => this.setState({isModifyingTitle: true})}
                       style={iconButtonStyle}
                     >
-                      <Icon name="pencil" />
+                      <Icon name="pencil"/>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: You have to insert a space before closing tag when the tag is single.

File: src/components/src/components/molecules/TTRenderer/index.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import React, { PropTypes } from 'react'
```
Comment (holenet) [inline src/components/src/components/molecules/TTRenderer/index.js]: What the this component???

File: src/components/molecules/TTRenderer/index.js [hunk 16]
```diff
@@ -0,0 +1,416 @@
...
+    this.drawGrid(ctx)
+    this.setHeaders(ctx)
+    this.drawSelections(ctx)
+
+    for (let i = 0; i < this.state.lectures.length; i += 1) {
+      this.drawLecture(ctx, this.state.lectures[i])
+    }
+  }
+
+  showStatus = () => {
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: This function seems just for debugging, so please remove it.

File: src/components/molecules/TTRenderer/index.js [hunk 17]
```diff
@@ -0,0 +1,416 @@
...
+    for (let j = 0; j <= 7; j += 1) {
+      ctx.moveTo((blockWidth + 2) * j, 0)
+      ctx.lineTo((blockWidth + 2) * j, canvasHeight)
+      ctx.stroke()
+    }
+  }
+
+  updateCanvas = () => {
+    const canvas = this.refs.canvas
+    const ctx = canvas.getContext('2d')
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: What the `ctx` means?
This name is hard to understand.

Comment (holenet) [reply src/components/molecules/TTRenderer/index.js]: If this is `context`, please use `context` as it is.
`ctx` is really unnecessary abbreviation.

File: src/components/molecules/TTRenderer/index.js [hunk 18]
```diff
@@ -0,0 +1,416 @@
...
+        if (this.overlap(i, j, topLeft, botRight)) { blocks[i][j] = 1 - blocks[i][j] }
+      }
+    }
+
+    this.setState({ blocks })
+
+    this.props.onChange(blocks)
+  }
+
+  getName = (name) => {
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: I cannot find any usages of this function.

File: src/components/molecules/LecturePopup/index.js (hunk 1 above)
Comment (Tommybro) [reply src/components/molecules/LecturePopup/index.js]: What this callBack do?

File: src/components/molecules/TTRenderer/index.js [hunk 19]
```diff
@@ -0,0 +1,416 @@
...
+      ctx.fillText(room, center.x - (ctx.measureText(room).width / 2), center.y + 20)
+    }
+  }
+
+  onClickCanvas = (e) => {
+    if (e.ctrlKey) { return }
+
+    window.document.removeEventListener('mousemove', this.onMouseMove)
+
+    const elt = ReactDOM.findDOMNode(this)
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: what is elt?
Please modify to understandable name

File: src/components/molecules/TTRenderer/index.js [hunk 20]
```diff
@@ -0,0 +1,416 @@
...
+    const rect = elt.getBoundingClientRect()
+    const eltLeft = rect.left
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: same here

File: src/components/molecules/TTRenderer/index.js [hunk 21]
```diff
@@ -0,0 +1,416 @@
...
+    const eltTop = rect.top
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: same here x2

File: src/components/molecules/TTRenderer/index.js [hunk 22]
```diff
@@ -0,0 +1,416 @@
...
+    const state = this.state
+    if (state.startPoint != null && state.endPoint != null) {
+      return `start: (${state.startPoint.x},${state.startPoint.y})`
+        + ' ' + `end: (${state.endPoint.x},${state.endPoint.y})`
+    }
+    return 'null'
+  }
+
+  overlap = (i, j, tl, br) => {
+    const tl1 = { x: (blockWidth + 2) * (j + 1), y: (blockHeight + 2) * (i + 1) }
```
Comment (Tommybro) [inline src/components/molecules/TTRenderer/index.js]: Please add comment above `overlap` that `tl1` is such as `topLeft of box 1
Or modify variable name understandable

File: src/components/src/components/molecules/TTRenderer/index.js (hunk 15 above)
Comment (Tommybro) [reply src/components/src/components/molecules/TTRenderer/index.js]: ㅋㅋㅋㅋ What are these 3 files under??
Already TTRenderer exists

**Comment by Tommybro** [comment]
Can you make each lecture has different color?

File: src/components/src/components/molecules/TTRenderer/index.js (hunk 15 above)
Comment (lameinthebox) [reply src/components/src/components/molecules/TTRenderer/index.js]: What do you mean?

Comment (lameinthebox) [reply src/components/src/components/molecules/TTRenderer/index.js]: There's no such line. I think that's outdated.

Comment (holenet) [reply src/components/src/components/molecules/TTRenderer/index.js]: @lameinthebox Look carefully the _path_ of this file.

File: src/components/molecules/TTRenderer/index.js (hunk 17 above)
Comment (lameinthebox) [reply src/components/molecules/TTRenderer/index.js]: Well, I didn't make it up. It's conventionally used when people use canvas. Do you still want to rename it?

Comment (holenet) [reply src/components/molecules/TTRenderer/index.js]: Um... Do as you want to do please.

File: src/components/molecules/TTRenderer/index.js (hunk 19 above)
Comment (lameinthebox) [reply src/components/molecules/TTRenderer/index.js]: Elt stands for element. I'm surprised you didn't know that.

Comment (holenet) [reply src/components/molecules/TTRenderer/index.js]: @lameinthebox ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

File: package.json [hunk 23]
```diff
@@ -91,6 +91,7 @@
     "redux": "^3.6.0",
     "redux-form": "^6.4.3",
     "redux-saga": "^0.14.1",
+    "semantic-ui": "^2.3.1",
```
Comment (holenet) [inline package.json]: Why did you add this package?

Comment (lameinthebox) [reply package.json]: Isn't it needed to use semantic ui?

File: src/components/molecules/TimeTable/index.js (hunk 11 above)
Comment (holenet) [reply src/components/molecules/TimeTable/index.js]: Do you mean this function is a skeleton?

File: package.json (hunk 23 above)
Comment (lameinthebox) [reply package.json]: I'm not sure where I got it

Comment (holenet) [reply package.json]: We were using `semantic-ui-react` instead of raw `semantic-ui`.
So, `semantic-ui` seems to be redundant package.

Comment (holenet) [reply package.json]: I guess `semantic-ui-react` includes `semantic-ui`.

Comment (lameinthebox) [reply package.json]: So can I delete that line?

**Comment by lameinthebox** [comment]
Now lectures have different colours.

screenshot from 2018-06-07 15-46-11

File: package.json (hunk 23 above)
Comment (holenet) [reply package.json]: I am not sure, too.
But, you can test if you can delete that line by following steps below;
1. remove the line `"semantic-ui": "^2.3.1",`
2. run command `$ npm install`
3. test the timetable
If it works well as before, you can remove that line.
Or, just leave it.

File: src/components/molecules/TimeTable/index.js (hunk 11 above)
Comment (lameinthebox) [reply src/components/molecules/TimeTable/index.js]: Whenever there's a change in the selected block set, `getBlocks` is called. Here, you can assign `blocks` to some other variable for further use (such as send it as a parameter for recommend).

File: package.json (hunk 23 above)
Comment (holenet) [reply package.json]: I modified comment

File: src/components/molecules/TimeTable/index.js (hunk 11 above)
Comment (holenet) [reply src/components/molecules/TimeTable/index.js]: Okay, I understand.

File: package.json (hunk 23 above)
Comment (lameinthebox) [reply package.json]: It works just fine without that line. I've removed it.

File: src/components/molecules/TimeTable/index.js [hunk 24]
```diff
@@ -231,13 +104,13 @@ class TimeTable extends React.Component {
           {this.props.id !== null &&
           <Menu.Item active fitted>
             {!this.state.isModifyingTitle ?
-              <div style={{ paddingLeft: 10, paddingRight: 10 }}>
+              <div style={{paddingLeft: 10, paddingRight: 10}}>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 25]
```diff
@@ -231,13 +104,13 @@ class TimeTable extends React.Component {
...
                 <h4>{this.state.title}
                   {this.props.canModify &&
                   <Popup
                     trigger={<button
                       className="ui icon button"
-                      onClick={() => this.setState({ isModifyingTitle: true })}
+                      onClick={() => this.setState({isModifyingTitle: true})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 26]
```diff
@@ -277,13 +150,13 @@ class TimeTable extends React.Component {
                   <Button
                     icon="add"
                     content="Add Lecture"
-                    style={{ ...iconButtonStyle, paddingLeft: 10, paddingRight: 10 }}
-                    onClick={() => this.setState({ searchOpen: true })}
+                    style={{...iconButtonStyle, paddingLeft: 10, paddingRight: 10}}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 27]
```diff
@@ -277,13 +150,13 @@ class TimeTable extends React.Component {
...
+                    onClick={() => this.setState({searchOpen: true})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 28]
```diff
@@ -277,13 +150,13 @@ class TimeTable extends React.Component {
...
                   />
                   {this.state.searchOpen &&
                   <SearchLecture
                     onAddLecture={this.props.onAddLecture}
-                    onClose={() => this.setState({ searchOpen: false })}
+                    onClose={() => this.setState({searchOpen: false})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 29]
```diff
@@ -293,15 +166,15 @@ class TimeTable extends React.Component {
             </Menu.Item>}
             {this.props.id !== null &&
             <Menu.Item active fitted>
-              <div style={{ paddingLeft: 10, paddingRight: 10 }}>
+              <div style={{paddingLeft: 10, paddingRight: 10}}>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 30]
```diff
@@ -293,15 +166,15 @@ class TimeTable extends React.Component {
...
                 {this.props.canCopyToMy &&
                 <Popup
                   trigger={<button
                     className="ui icon button"
                     onClick={() => this.props.onCopyToMy(this.props.id)}
                     style={iconButtonStyle}
                   >
-                    <Icon name="copy" />
+                    <Icon name="copy"/>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 31]
```diff
@@ -313,18 +186,18 @@ class TimeTable extends React.Component {
                     onClick={() => this.props.onBookmark(this.props.id)}
                     style={iconButtonStyle}
                   >
-                    <Icon name="bookmark" />
+                    <Icon name="bookmark"/>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 32]
```diff
@@ -313,18 +186,18 @@ class TimeTable extends React.Component {
...
                   </button>}
                   content="Bookmark this timetable"
                   inverted
                 />
                 <Popup
                   trigger={<button
                     className="ui icon button"
-                    onClick={() => this.setState({ isSending: true })}
+                    onClick={() => this.setState({isSending: true})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 33]
```diff
@@ -313,18 +186,18 @@ class TimeTable extends React.Component {
...
                     style={iconButtonStyle}
                   >
-                    <Icon name="send" />
+                    <Icon name="send"/>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 34]
```diff
@@ -343,23 +216,23 @@ class TimeTable extends React.Component {
                       />
                     </Form> :
                     'Send this timetable to other student'}
-                  onClose={() => this.setState({ isSending: false })}
+                  onClose={() => this.setState({isSending: false})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 35]
```diff
@@ -343,23 +216,23 @@ class TimeTable extends React.Component {
...
                   on={this.state.isSending ? 'click' : 'hover'}
                   inverted={!this.state.isSending}
                 />
                 {this.props.canDelete &&
                 <Popup
                   trigger={<button
                     className="ui icon button"
-                    onClick={() => this.setState({ isDeleting: true })}
+                    onClick={() => this.setState({isDeleting: true})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 36]
```diff
@@ -343,23 +216,23 @@ class TimeTable extends React.Component {
...
                     style={iconButtonStyle}
                   >
-                    <Icon name="trash" color="red" />
+                    <Icon name="trash" color="red"/>
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 37]
```diff
@@ -343,23 +216,23 @@ class TimeTable extends React.Component {
...
                   </button>}
                   content={this.state.isDeleting ?
-                    <Button color="red" content="Delete" onClick={() => this.props.onDeleteTimeTable(this.props.id)} /> :
+                    <Button color="red" content="Delete" onClick={() => this.props.onDeleteTimeTable(this.props.id)}/> :
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js [hunk 38]
```diff
@@ -343,23 +216,23 @@ class TimeTable extends React.Component {
...
                     'Delete this timetable'}
-                  onClose={() => this.setState({ isDeleting: false })}
+                  onClose={() => this.setState({isDeleting: false})}
```
Comment (holenet) [inline src/components/molecules/TimeTable/index.js]: Please follow the ESLint rule.

File: src/components/molecules/TimeTable/index.js (hunk 24 above)
Comment (holenet) [reply src/components/molecules/TimeTable/index.js]: Here.

File: src/components/molecules/LecturePopup/index.js (hunk 1 above)
Comment (holenet) [reply src/components/molecules/LecturePopup/index.js]: Please leave a comment here.

File: src/components/molecules/TTRenderer/index.js [hunk 39]
```diff
@@ -0,0 +1,364 @@
...
+          ctx.fillRect(start.x, start.y, blockWidth, blockHeight)
+        }
+      }
+    }
+  }
+
+  drawLecture = (ctx, lecture) => {
+
+    const days = ['헤더', '월', '화', '수', '목', '금', '토']
+    const colors = ['rgb(0,116,217)', 'rgb(255,220,0)', 'rgb(46,204,64)']
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: Are there only 3 colors?
Does it mean that at least two lectures among four lectures have same color?

Comment (lameinthebox) [reply src/components/molecules/TTRenderer/index.js]: Feel free to suggest colours. I'll append them.

Comment (holenet) [reply src/components/molecules/TTRenderer/index.js]: I suggest to use continuous color set by HSL coloring.
You can refer to [here](https://www.w3schools.com/colors/colors_hsl.asp).
If you fix(or narrow the range) value of S and L, and pick H by hash of course's name, the colors of lectures are distinct(maybe) and have consistency color scheme.

Comment (lameinthebox) [reply src/components/molecules/TTRenderer/index.js]: That seems pretty nice! Thanks.

File: src/components/molecules/TTRenderer/index.js [hunk 40]
```diff
@@ -0,0 +1,369 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+
+import LecturePopup from '../../../containers/LecturePopup'
+
+
+const blockWidth = 154
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: Could you change the code so that the timetable fill the parent container(component)?
That means, there should be no space no matter what value the screen width have.

File: src/components/molecules/TTRenderer/index.js [hunk 41]
```diff
@@ -5,10 +5,10 @@ import PropTypes from 'prop-types'
...
 
 
-const blockWidth = 154
-const blockHeight = 29
-const canvasWidth = (blockWidth + 2) * 7
-const canvasHeight = (blockHeight + 2) * 25
+var blockWidth = 154
+var blockHeight = 29
+var canvasWidth = (blockWidth + 2) * 7
+var canvasHeight = (blockHeight + 2) * 25
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: Please use `let` not `var`.

File: src/components/molecules/TTRenderer/index.js [hunk 42]
```diff
@@ -42,6 +45,24 @@ class TTRenderer extends React.Component {
     this.updateCanvas()
   }
 
+  updateWindowDimensions = () => {
+
+    if (1200 <= window.innerWidth) {
+      blockWidth = 154
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: Why is the `blockWidth` discontinuous?
I cannot grab the intent of these `if else`s.

File: src/components/molecules/TTRenderer/index.js [hunk 43]
```diff
@@ -256,8 +277,16 @@ class TTRenderer extends React.Component {
 
   updateCanvas = () => {
     const canvas = this.refs.canvas
+    if (!canvas) {
+        return 
+    }
+
     const ctx = canvas.getContext('2d')
-    ctx.clearRect(0, 0, canvasWidth, canvasHeight)
+    //ctx.clearRect(0, 0, canvasWidth, canvasHeight)
```
Comment (holenet) [inline src/components/molecules/TTRenderer/index.js]: Please delete this line.

File: src/components/molecules/TTRenderer/index.js (hunk 42 above)
Comment (lameinthebox) [reply src/components/molecules/TTRenderer/index.js]: I've manually found the best value of blockWidth for each case.
